src/modules/map/components/CompareSlider.vue

```diff
@@ -1,35 +1,71 @@
-<template>
-    <div class="compare-slider">
-        <font-awesome-icon
-            class="compare-slider-caret-left"
-            :icon="['fas', 'caret-up']"
-            :rotation="270"
-        />
-        <div class="compare-slider-line"></div>
-
-        <font-awesome-icon
-            class="compare-slider-caret-right"
-            :icon="['fas', 'caret-up']"
-            :rotation="90"
-        />
-    </div>
-</template>
-
-<script>
+<script setup>
 import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
+import { computed, defineProps, onMounted, ref, toRefs, watch } from 'vue'
+import { useStore } from 'vuex'
 
-export default {
-    components: {
-        FontAwesomeIcon,
+const props = defineProps({
+    clientWidth: {
+        type: Number,
+        default: window.innerWidth,
     },
+})
+
+const { clientWidth } = toRefs(props)
+
+const compareRatio = ref(-0.5)
+
+const store = useStore()
+const storeCompareRatio = computed(() => store.state.ui.compareRatio)
+const compareSliderPosition = computed(() => compareRatio.value * 100 + '%')
+
+watch(storeCompareRatio, (newValue) => {
+    compareRatio.value = newValue
+})
+
+onMounted(() => {
+    compareRatio.value = storeCompareRatio.value
+})
+function generateCSSPosition() {
+    return { left: compareSliderPosition.value }
+}
```

Comment:
`compareSliderPosition` is only used to create this CSS style object, maybe we could create this object directly in the computed `compareSliderPosition` and remove this function.